### PR TITLE
Allows the `v` char in the version string

### LINKED ISSUE #5342: Operator SDK doesn't support `v` char in version, claiming it's not semantic
#### What did you do?

Running `make bundle` the Operator SDK fails with fatal error.

#### What did you expect to see?

I expected a bundle should build fine with following as version: `v0.2.0-13-ga74dac1`

#### What did you see instead? Under which circumstances?

The following error has been raised:

```
FATA[0000] invalid command options: v0.2.0-13-ga74dac1 is not a valid semantic version: Invalid character(s) found in major number "v0"
```

#### Environment

**Operator type:**

/language go

**Kubernetes cluster type:**

Minikube v1.23.2

```sh
$ operator-sdk version
operator-sdk version: "v1.14.0", commit: "78f08b4852faf344ad3ef457c54f86087aaa0a0a", kubernetes version: "1.21", go version: "go1.16.9", GOOS: "linux", GOARCH: "amd64"

$ go version
go version go1.16 linux/amd64

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.2", GitCommit:"8b5a19147530eaac9476b0ab82980b4088bbc1b2", GitTreeState:"clean", BuildDate:"2021-09-15T21:32:41Z", GoVersion:"go1.16.8", Compiler:"gc", Platform:"linux/amd64"}
```

#### Possible Solution

I think the Operator SDK should be adjusted to allow using a `v` char in version string, as that is extremely common. After https://semver.org/#is-v123-a-semantic-version

### PR DESCRIPTION
**Pull request by cardil**
**Description of the change:**

Fixes #5342 

**Checklist**

If the pull request includes user-facing changes, extra documentation is required:
- [x] Add a new changelog fragment in `changelog/fragments` (see [`changelog/fragments/00-template.yaml`](https://github.com/operator-framework/operator-sdk/tree/master/changelog/fragments/00-template.yaml))
- [ ] Add or update relevant sections of the docs website in [`website/content/en/docs`](https://github.com/operator-framework/operator-sdk/tree/master/website/content/en/docs)

### CODE REVIEW

**Review by varshaprasad96 — APPROVED** [review]
/lgtm

File: internal/cmd/operator-sdk/generate/internal/genutil.go
Comment (camilamacedo86) [inline internal/cmd/operator-sdk/generate/internal/genutil.go:40]: Just a nit. It seems to be missing the entry in https://github.com/operator-framework/operator-sdk/blob/master/changelog/fragments/00-template.yaml.

**Comment by asmacdo** [comment]
@cardil thanks for the PR! Once the changelog fragment is added, we should be good to merge this :)

File: internal/cmd/operator-sdk/generate/internal/genutil.go
Comment (asmacdo) [reply internal/cmd/operator-sdk/generate/internal/genutil.go:40]: @cardil The changelog fragments get consumed and put [here](https://github.com/operator-framework/operator-sdk/tree/master/changelog/generated) each time we cut a release.

**Comment by joelanford** [comment]
Before we merge this, I just wanted to double check a few things.
1. Are we talking about the version derived from `csv.spec.version`?
2. If so, are we sure OLM supports the version that includes a `v` prefix?


/hold

(feel free to unhold if the above questions are sufficiently answered)

File: internal/cmd/operator-sdk/generate/internal/genutil.go
Comment (cardil) [reply internal/cmd/operator-sdk/generate/internal/genutil.go:40]: Makes 100% sense. Although, this simple explanation should be stated in the docs.

File: internal/cmd/operator-sdk/generate/internal/genutil.go
Comment (varshaprasad96) [inline internal/cmd/operator-sdk/generate/internal/genutil.go:38]:
```suggestion
	v, err := semver.ParseTolerant(version)
```
`semver.ParseTolerant` (https://github.com/blang/semver/blob/af3461a9cbcf1f3f5889d21b83f5ef63880c33a8/v4/semver.go#L240) would sanitize the string for us.

**Review by varshaprasad96 — COMMENTED** [review]
Based on what Joe said, even if `csv.Spec.Version` accepts string with `v`, I think with this change we would still error here: https://github.com/operator-framework/operator-sdk/blob/43b5394cf5f5b4482e242d9369559b9fe345400b/internal/generate/clusterserviceversion/clusterserviceversion.go#L161

I think olm may accept `spec.Version` with `v` prefixed. Looks like `csv.spec.Version` is of type [`OperatorVersion`](https://github.com/operator-framework/api/blob/160240770853d11a62b3ff721b3ffed2d8edf06d/pkg/lib/version/version.go#L13), and atleast while Umarshalling it they [sanitize](https://github.com/operator-framework/api/blob/160240770853d11a62b3ff721b3ffed2d8edf06d/pkg/lib/version/version.go#L50) the string. Nevertheless, it would be helpful to have a test to verify if we need to change `semver.Parse()` to `semver.ParseTolerant()` while generating csv with this change.

File: internal/cmd/operator-sdk/generate/internal/genutil.go
Comment (cardil) [reply internal/cmd/operator-sdk/generate/internal/genutil.go:39]: I reverted this change. The test was failing with this, complaining the version do not match.

**Comment by cardil** [comment]
At @joelanford:

With the latest changes to this PR, I'm removing the `v` prefix if there is any. This means any other behavior of _operator-sdk_ is unchanged.

File: changelog/fragments/allow-v-in-version.yaml
Comment (joelanford) [inline changelog/fragments/allow-v-in-version.yaml:7]: Could you expand on (a) which versions are allowed to have the `v` prefix and (b) what commands will translate those versions to non-`v`-prefixed versions and (c) which output artifacts will drop the `v` prefix and which won't?

IIRC there's a handful of places where versions are specified by users and a separate handful where versions are written/pushed by the `operator-sdk` commands or the project Makefile.

Comment (cardil) [reply changelog/fragments/allow-v-in-version.yaml]: Hope the new description is more informative. Please, re-evaluate @joelanford.

Comment (cardil) [reply changelog/fragments/allow-v-in-version.yaml]: @joelanford please take a look once again. Thanks!

**Comment by cardil** [comment]
Ping!

**Comment by cardil** [comment]
Can I get re-review?

**Comment by varshaprasad96** [comment]
/approve
/lgtm
/ok-to-test
The change looks good to me. @joelanford any thoughts?

**Pull request closed without merging**
